packages/wouter/src/use-browser-location.js

```diff
@@ -54,12 +54,18 @@ export const useBrowserLocation = (opts = {}) => [
   useEvent((to, navOpts) => navigate(absolutePath(to, opts.base), navOpts)),
 ];
 
+const patchKey = Symbol.for("wouter_v3");
+
 // While History API does have `popstate` event, the only
 // proper way to listen to changes via `push/replaceState`
 // is to monkey-patch these methods.
 //
 // See https://stackoverflow.com/a/4585031
-if (typeof history !== "undefined") {
+if (typeof history !== "undefined" && typeof window[patchKey] === "undefined") {
+  // patch history object only once
+  // See: https://github.com/molefrog/wouter/issues/167
+  Object.defineProperty(window, patchKey, { value: true });
```

Comment:
Not a big deal, but shouldn't this property be set after the actual patch? I mean if the code below somehow fails (exception) the patch will not be complete.

Reply:
yes, it should be nice